src/pages/UpdateProduct.js

```diff
@@ -2,72 +2,23 @@ import React, { useState, useEffect } from 'react';
 import axios from 'axios';
 import './styles.css';
 
-const UpdateProduct = ({ productId }) => {
-  const [name, setName] = useState('');
-  const [price, setPrice] = useState('');
-  const [description, setDescription] = useState('');
-  const [category, setCategory] = useState('');
-  const [company, setCompany] = useState('');
-  const [featured, setFeatured] = useState(false);
-  const [freeShipping, setFreeShipping] = useState(false);
-  const [inventory, setInventory] = useState(15);
+const UpdateProduct = ({ product, onUpdate }) => {
+  const [updatedProductData, setUpdatedProductData] = useState({...product});
   const [message, setMessage] = useState('');
 
-  useEffect(() => {
-    const fetchProduct = async () => {
-      try {
-        const response = await axios.get(
-          `https://ecommerce-6kwa.onrender.com/api/v1/products/${productId}`,
-          { withCredentials: true }
-        );
-
-        // The product endpoint returns an object nested under the field 'product' eg: { "product" : { name: blah, price: blah } } . So we need to add `.product` here after response.data. 
-        //An alternative approach would be to use object destructuring and do { product } = response.data
-        const product = response.data.product;
-
-      setName(product.name);
-      setPrice(product.price);
-      setDescription(product.description);
-      setCategory(product.category);
-      setCompany(product.company);
-      setFeatured(product.featured);
-      setFreeShipping(product.freeShipping);
-      setInventory(product.inventory);
-
-
-
-      } catch (error) {
-        console.log('Error fetching product:', error);
-      }
-    };
-
-    fetchProduct();
-  }, [productId]);
-
   const handleSubmit = async (e) => {
     e.preventDefault();
 
-    const product = {
-      name,
-      price,
-      description,
-      category,
-      company,
-      featured,
-      freeShipping,
-      inventory,
-    };
-
-
-
     try {
       const response = await axios.patch(
         `https://ecommerce-6kwa.onrender.com/api/v1/products/${productId}`,
-        product,
+        updatedProductData,
         { withCredentials: true }
       );
       console.log('Product updated successfully:', response.data);
       setMessage('Product updated successfully');
+      setUpdatedProductData(response?.data?.product)
+      onUpdate()
```

Comment:
Call the callback function that was passed in via props, to let the parent component (in this case `SingleProductPage`) know it should reload the product info since it has been updated.